### DISCUSSION TITLE
Add a feature that allows scripts to be pinned or locked

### DISCUSSION THREAD

**Discussion by living2room** [godotengine/godot-proposals#9574] [Editor]
I recommend adding a feature to the script editor that allows scripts to be pinned or locked. 

The pinned scripts would always stay at the top position. 

With the locking feature, the script would not switch automatically when other nodes are clicked, unless the script is attached to the node being clicked, in which case it would still switch and unlock automatically. Also, locked scripts would not be closed when selecting ‘Close All’ from the right-click context menu

**Comment by Bluedisaster** [comment]
I would really like to have this feature. I have a _todo.txt and some other scripts like utils.gd and global.gd, which I always use. But the list is polluted by script I don't need every time and currently I use "Close all but this", but then I have to open my needed files always again.

**Comment by arnzzz** [comment]
Just adding my support for this idea as well. 

I was feeling like I was constantly fighting with the script panel, but on a closer look iv'e realized its this issue precisely. It presents mostly when i'm using any autoloaded scripts as i'm always back and fourth into those, it would be so nice to pin them at the top of the list.

I wish I was more capable to submit some code myself to do this, seems like it might be a cool little project. (in fact im gonna take a look how to contribute, but dont hold your breath) :)

**Comment by arnzzz** [comment]
I made a little EditorPlugin with a timer just to test things out.

The script panel is updating a lot behind the scenes, so anything I do is periodically getting overridden.

Every 3 seconds i check the script panels ItemList for a hard coded named file, then i move that to the top of the list using ItemList.move_item(from, to).

But whatever is going on behind the scenes in the editor is messing with it, moving items in that list around without my input.

I disabled 'script temperature' and 'sort scripts by' in the Editor Settings but its still doing something I have no access to with my limited knowledge. sorry xx
